### PR TITLE
#USDU-155 Create import/export helpers to clean the UsdMenu 

### PR DESCRIPTION
**Pull request by judubu**
## Purpose of this PR

**Ticket/Jira #:** https://jira.unity3d.com/browse/USDU-155

Preparatory work for the mesh import/conversion layer fixes.
UsdMenu used to contain a lot of import logic making it hard to reuse and test. All the logic has been moved to 2 new static classes ImportHelpers and ExportHelpers.

## Testing

**Functional Testing status:**
Tried to cover 100% of the Import/ExportHelpers. 

**Performance Testing status:**
N/A

## Overall Product Risks

**Complexity:**
Low

**Halo Effect:**
Low

## Additional information

**Note to reviewers:**

**Reminder:**

- [ ] Add entry in CHANGELOG.md _(if applicable)_

### CODE REVIEW

File: package/com.unity.formats.usd/Editor/Scripts/Behaviors/UsdMenu.cs [hunk 1]
```diff
@@ -21,50 +21,8 @@
 
 namespace Unity.Formats.USD
 {
-    public class UsdMenu : MonoBehaviour
+    public class UsdMenu
```
Comment (clusty) [inline package/com.unity.formats.usd/Editor/Scripts/Behaviors/UsdMenu.cs]: make the class static maybe ?

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using USD.NET;
+
+namespace Unity.Formats.USD
+{
+    public static class ExportHelpers
+    {
+#if UNITY_EDITOR
+        public static Scene InitForSave(string defaultName, string fileExtension = "usd,usda,usdc")
+        {
+            var filePath = EditorUtility.SaveFilePanel("Export USD File", "", defaultName, fileExtension);
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs]: I would split this function differently:
The only editor only function is EditorUtility.SaveFilePanel: add that in the editor assembly and keep the rest of of this function as a runtime function.

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+#endif
+
+        public static void ExportSelected(BasisTransformation basisTransform,
+            string fileExtension = "usd,usda,usdc",
+            bool exportMonoBehaviours = false)
+        {
+            Scene scene = null;
+
+            foreach (GameObject go in Selection.gameObjects)
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs]: Selection exists only in the editor. Make a function that exports a list of GOs and call that with the selection from the editor assembly.

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            }
+
+            try
+            {
+                UsdToGameObject(root, scene, importOptions);
+                return root;
+            }
+            catch (SceneImporter.ImportException)
+            {
+                Object.DestroyImmediate(root);
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs:53]: Not sure if it makes a difference: in playmode you wanna call Destroy, and out of playmode DestroyImmediate. Make yourself an utility function that abstracts this.

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+                return null;
+            }
+        }
+
+#if UNITY_EDITOR
+        public static void ImportAsPrefab(Scene scene)
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs]: Return the import results.

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+                SceneImporter.SavePrefab(go, prefabPath, clipName, importOptions);
+            }
+            finally
+            {
+                Object.DestroyImmediate(go);
+                scene.Close();
+            }
+        }
+
+        public static void ImportAsTimelineClip(Scene scene)
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs]: Return the import result.

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+            // them all, to avoid confusion.
+            return pxr.SdfPath.AbsoluteRootPath();
+        }
+
+        /// <summary>
+        /// Returns the selected object path or "Assets/" if no object is selected.
+        /// </summary>
+        static string GetSelectedAssetPath()
+        {
+            Object[] selectedAsset = Selection.GetFiltered(typeof(Object), SelectionMode.Assets);
```
Comment (clusty) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs:124]: Think Selection is Editor only

**Review by clusty — APPROVED** [review]

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs (hunk 2 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs]: Fixed in 3cd29eb4

File: package/com.unity.formats.usd/Editor/Scripts/Behaviors/UsdMenu.cs (hunk 1 above)
Comment (judubu) [reply package/com.unity.formats.usd/Editor/Scripts/Behaviors/UsdMenu.cs]: Fixed in 3cd29eb4

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs (hunk 3 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ExportHelpers.cs]: Fixed in 3cd29eb4

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs (hunk 5 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs]: Fixed in 53a493cd

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs (hunk 6 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs]: Fixed in 53a493cd

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs (hunk 7 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Scene/ImportHelpers.cs:124]: Fixed 53a493cd

**Review by mfe — APPROVED** [review]
LGTM
Did some testing on 2020.2 (win), change looks indeed transparent.

**Review by clusty — APPROVED** [review]

**Review by mfe — APPROVED** [review]
LGTM

**Pull request merged**